pmoves/tools/observability/llm_observability_specialist.py

```
@@ -0,0 +1,369 @@
+            avg(latency) as avg_latency_ms,
+            quantile(0.50)(latency) as p50_latency_ms,
+            quantile(0.95)(latency) as p95_latency_ms,
+            quantile(0.99)(latency) as p99_latency_ms,
+            sum(prompt_tokens) as total_prompt_tokens,
+            sum(completion_tokens) as total_completion_tokens
+        FROM requests
+        WHERE start_time >= now() - INTERVAL {hours} HOUR
+          {model_filter}
+        GROUP BY model_name
+        ORDER BY total_requests DESC
+        """
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)